core/src/smartcontracts/isi/asset.rs

```diff
@@ -172,7 +170,11 @@ pub mod isi {
                 *quantity = quantity
                     .checked_add(mint.object)
                     .ok_or(MathError::Overflow)?;
-                metrics_arc.tx_amounts.observe((*quantity).into_metric());
+                #[allow(clippy::float_arithmetic)]
```

Comment:
This doesn't look particularly efficient. How is this better than what we had before?

Reply:
Because previously you could poll the metrics and recieve information that includes a partially applied block. Doing it this way keeps the data local to a thread and it updates the metrics at a predictable time.

Reply:
Why couldn't you update the information when the data is requested instead? Lazy evaluation, allows you not to pay for metrics, when you don't use them, at the cost of doing more work when you do.

Reply:
Thread locality sounds good. Given that we aggregate data, should be fine in general.